src/map/intif.c

```diff
@@ -439,6 +439,97 @@ int intif_request_registry(struct map_session_data *sd, int flag)
 	return 0;
 }
 
+//=================================================================
+// Account Storage
+//-----------------------------------------------------------------
+
+/**
+ * Request the inter-server for a character's storage data.
+ * @packet 0x3010  [out] <account_id>.L
+ * @param  sd      [in]  pointer to session data.
+ */
+void intif_request_account_storage(const struct map_session_data *sd)
+{
+	nullpo_retv(sd);
+
+	/* Check for character server availability */
+	if (intif->CheckForCharServer())
+		return;
+
+	WFIFOHEAD(inter_fd, 6);
+	WFIFOW(inter_fd, 0) = 0x3010;
+	WFIFOL(inter_fd, 2) = sd->status.account_id;
+	WFIFOSET(inter_fd, 6);
+}
+
+/**
+ * Parse the reception of account storage from the inter-server.
+ * @packet 0x3805 [in] <packet_len>.W <account_id>.L <struct item[]>.P
+ * @param  fd     [in] file/socket descriptor.
+ */
+void intif_parse_account_storage(int fd)
+{
+	int account_id = 0, payload_size = 0, storage_count = 0;
+	int i = 0;
+	struct map_session_data *sd = NULL;
+
+	Assert_retv(fd > 0);
+
+	if ((payload_size = RFIFOW(fd, 2) - 8) == 0)
+		return; // no data, nothing to do.
```

Comment:
If there's no data, we likely still want to know whose it was, and mark the storage data as received for the character